File: Leetcoding-Actions/LC-Monthly-Challenges/2020-11/10-Flip-an-Image.java
class Solution {
    public int[][] flipAndInvertImage(int[][] A) {
        /*
        T: O(m * n) and S: O(1)
        */
        int nrow = A.length;
        int ncol = A[0].length;
        
        for (int row = 0; row < nrow; row++) {
            int i = 0;
            int j = ncol - 1;
            while (i < j) {
                int temp = A[row][i];
                A[row][i] = A[row][j];
                A[row][j] = temp;
                i++; j--;
            }
        }
        
        for (int i = 0; i < nrow; i++) {
            for (int j = 0; j < ncol; j++) {
                A[i][j] = 1 - A[i][j];
            }
        }
        return A;
    }
}

File: Leetcoding-Actions/LC-Monthly-Challenges/2020-11/10-Flip-an-Image.py
class Solution:
    def flipAndInvertImage(self, A: List[List[int]]) -> List[List[int]]:
        '''
        T: O(m * n) and S: O(1)
        '''
        if not A: return A
        
        nrow, ncol = len(A), len(A[0])
        
        for row in range(nrow):
            i, j = 0, ncol - 1
            while i < j:
                A[row][i], A[row][j] = A[row][j], A[row][i]
                i, j = i + 1, j - 1
        
        for i in range(nrow):
            for j in range(ncol):
                A[i][j] = 1 - A[i][j]
                
        return A
